test/Microsoft.IdentityModel.TestUtils/Default.cs

```diff
@@ -407,6 +407,48 @@ public static List<Claim> PayloadClaims
             };
         }
 
+        public static List<Claim> PayloadJsonClaims
+        {
+            get => new List<Claim>
+                    {
+                        new Claim(JwtRegisteredClaimNames.Aud, Audience, ClaimValueTypes.String),
+                        new Claim(JwtRegisteredClaimNames.Iss, Issuer, ClaimValueTypes.String),
+                        new Claim("ClaimValueTypes.String", "ClaimValueTypes.String.Value", ClaimValueTypes.String),
+                        new Claim("ClaimValueTypes.Boolean.true", "true", ClaimValueTypes.Boolean),
+                        new Claim("ClaimValueTypes.Boolean.false", "false", ClaimValueTypes.Boolean),
+                        new Claim("ClaimValueTypes.Double", "123.4", ClaimValueTypes.Double),
+                        new Claim("ClaimValueTypes.DateTime.IS8061", "2019-11-15T14:31:21.6101326Z", ClaimValueTypes.DateTime),
+                        new Claim("ClaimValueTypes.DateTime", "2019-11-15", ClaimValueTypes.DateTime),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.Json1", @"{""jsonProperty1"":""jsonvalue1""}", System.IdentityModel.Tokens.Jwt.JsonClaimValueTypes.Json),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.Json2", @"{""jsonProperty2"":""jsonvalue2""}", System.IdentityModel.Tokens.Jwt.JsonClaimValueTypes.Json),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.JsonNull", "", System.IdentityModel.Tokens.Jwt.JsonClaimValueTypes.JsonNull),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.JsonArray1", @"[1,2,3]", System.IdentityModel.Tokens.Jwt.JsonClaimValueTypes.JsonArray),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.JsonArray2", @"[1,""2"",3]", System.IdentityModel.Tokens.Jwt.JsonClaimValueTypes.JsonArray),
+                        new Claim("ClaimValueTypes.JsonClaimValueTypes.Integer1", "1", ClaimValueTypes.Integer),
+                    };
+        }
+
+        public static Dictionary<string, object> PayloadJsonDictionary
+        {
+            get => new Dictionary<string, object>()
+            {
+                { JwtRegisteredClaimNames.Aud, Audience },
+                { JwtRegisteredClaimNames.Iss, Issuer },
+                { "ClaimValueTypes.String", "ClaimValueTypes.String.Value" },
+                { "ClaimValueTypes.Boolean.true", true },
+                { "ClaimValueTypes.Boolean.false", false },
+                { "ClaimValueTypes.Double", 123.4 },
+                { "ClaimValueTypes.DateTime.IS8061", DateTime.TryParse("2019-11-15T14:31:21.6101326Z", out DateTime dateTimeValue1) ? dateTimeValue1 : new DateTime()},
```

Comment:
1. While using claims from dictionary to create a token:

```
- DateTime.TryParse("2019-11-15T14:31:21.6101326Z") result is 11/15/2019 6:31:21 AM( which will be the value of the Claim after converting dictionary to Claim list)
- 11/15/2019 6:31:21 AM after serialization becomes 2019-11-15T06:31:21
- 2019-11-15T06:31:21 after decoding the token becomes 2019-11-15T14:31:21.0000000Z
```

2. While using Subject.Claims to create a token:

`- "2019-11-15T14:31:21.6101326Z" is the value of the Claim and remains same after decoding the token.`

Reply:
Following up, found that multiple dictionary to claim conversions is the reason for this. Will be adding fix in next commit.